packages/suite/src/hooks/suite/usePinModal.ts

```diff
@@ -0,0 +1,28 @@
+import { selectDeviceButtonRequests } from '@suite-common/wallet-core';
+import { useSelector } from 'src/hooks/suite';
+
+export const usePin = () => {
+    const buttonRequests = useSelector(selectDeviceButtonRequests);
+
+    const pinRequestType = buttonRequests[buttonRequests.length - 1];
+    const invalidCounter = buttonRequests.filter(r => r.code === 'ui-invalid_pin').length || 0;
+
+    const isExtended =
+        (pinRequestType?.code &&
+            ['PinMatrixRequestType_NewFirst', 'PinMatrixRequestType_NewSecond'].includes(
```

Comment:
this array can be extracted as constant and named properly to explain by code why those two codes are important together

Reply:
for example something like: `REQUEST_NEW_PIN_CODES`